src/analysis/specific_limits.lean

```diff
@@ -351,6 +406,43 @@ end
 
 end geometric
 
+section mul_geometric
+
+lemma summable_norm_pow_mul_geometric_of_norm_lt_1 {R : Type*} [normed_ring R]
+  (k : ℕ) {r : R} (hr : ∥r∥ < 1) : summable (λ n : ℕ, ∥(n ^ k * r ^ n : R)∥) :=
+begin
+  rcases exists_between hr with ⟨r', hrr', h⟩,
+  exact summable_of_is_O_nat _ (summable_geometric_of_lt_1 ((norm_nonneg _).trans hrr'.le) h)
+    (is_o_pow_const_mul_const_pow_const_pow_of_norm_lt _ hrr').is_O.norm_left
+end
+
+lemma summable_pow_mul_geometric_of_norm_lt_1 {R : Type*} [normed_ring R] [complete_space R]
+  (k : ℕ) {r : R} (hr : ∥r∥ < 1) : summable (λ n, n ^ k * r ^ n : ℕ → R) :=
+summable_of_summable_norm $ summable_norm_pow_mul_geometric_of_norm_lt_1 _ hr
+
+lemma has_sum_coe_mul_geometric_of_norm_lt_1 {𝕜 : Type*} [normed_field 𝕜] [complete_space 𝕜]
+  {r : 𝕜} (hr : ∥r∥ < 1) : has_sum (λ n, n * r ^ n : ℕ → 𝕜) (r / (1 - r) ^ 2) :=
+begin
+  have A : summable (λ n, n * r ^ n : ℕ → 𝕜),
+    by simpa using summable_pow_mul_geometric_of_norm_lt_1 1 hr,
+  have B : has_sum (pow r : ℕ → 𝕜) (1 - r)⁻¹, from has_sum_geometric_of_norm_lt_1 hr,
+  refine A.has_sum_iff.2 _,
+  have hr' : r ≠ 1, by { rintro rfl, simpa [lt_irrefl] using hr },
+  rw [pow_two, ← div_div_eq_div_mul, eq_div_iff (sub_ne_zero.2 hr'.symm), mul_sub, mul_one,
```

Comment:
Since the proof is not completely trivial (there is a trick here) and a sequence of rewrites is not easy to follow, could you either convert it to a more readable proof with a calc block, or explain with comments what is going on?